workflows4s-core/src/main/scala/workflows4s/wio/internal/ExecutionProgressEvaluator.scala

```diff
@@ -143,6 +144,13 @@ object ExecutionProgressEvaluator {
     override def onRecovery[Evt](wio: WIO.Recovery[Ctx, In, Err, Out, Evt]): WIOExecutionProgress[WCState[Ctx]] =
       WIOExecutionProgress.Recovery(result)
 
+    override def onForEach[ElemId, InnerCtx <: WorkflowContext, ElemOut <: WCState[InnerCtx], InterimState <: WCState[Ctx]](
+        wio: WIO.ForEach[Ctx, In, Err, Out, ElemId, InnerCtx, ElemOut, InterimState],
+    ): WIOExecutionProgress[WCState[Ctx]] = {
+      val elemModel = ExecProgressVisitor(wio.elemWorkflow, None, None, None).run.toModel
+      WIOExecutionProgress.ForEach(result, elemModel, Map(), wio.meta) // TODO empty map!!!
```

Comment:
TODO

Should we just save incorporated states? Or a tuple or raw inner stated + incorporated?